oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationToken.java

```diff
@@ -36,13 +36,27 @@
 public class JwtAuthenticationToken extends AbstractOAuth2TokenAuthenticationToken<Jwt> {
 	private static final long serialVersionUID = SpringSecurityCoreVersion.SERIAL_VERSION_UID;
 
+	private String name;
+
 	/**
 	 * Constructs a {@code JwtAuthenticationToken} using the provided parameters.
 	 *
 	 * @param jwt the JWT
 	 */
 	public JwtAuthenticationToken(Jwt jwt) {
 		super(jwt);
+		this.name = this.getToken().getSubject();
+	}
+
+	/**
+	 * Constructs a {@code JwtAuthenticationToken} using the provided parameters.
+	 *
+	 * @param jwt  the JWT
+	 * @param name principal name
+	 */
+	public JwtAuthenticationToken(Jwt jwt, String name) {
```

Comment:
I think we can live without this constructor for now. We want to be careful to not get too many constructors--
we'd probably eventually introduce a builder instead of this constructor.

Just adding `(Jwt, Collection, String)` should be fine for this PR, unless you have a concern with leaving this constructor out.